src/agent/advertising_proxy.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ *    Copyright (c) 2020, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definition for Advertising Proxy.
+ */
+
+#ifndef OTBR_ADVERTISING_PROXY_HPP_
+#define OTBR_ADVERTISING_PROXY_HPP_
+
+#if OTBR_ENABLE_ADVERTISING_PROXY
+
+#include <stdint.h>
+
+#include <openthread/instance.h>
+#include <openthread/srp_server.h>
+
+#include "mdns/mdns.hpp"
+
+namespace otbr {
+
+/**
+ * This class implements the Advertising Proxy.
+ *
+ */
+class AdvertisingProxy
+{
+public:
+    explicit AdvertisingProxy(Mdns::Publisher *aPublisher);
+
+    otbrError Start(otInstance *aInstance);
+
+    void Stop();
+
+private:
+    struct OutstandingUpdate
+    {
+        const otSrpServerHost *mHost;  // The host represents a collection of updates.
```

Comment:
```suggestion
        const otSrpServerHost *mHost;  ///< The host represents a collection of updates.
```

Reply:
Those are private members/definitions, and are not expected to be in the doxygen. Actually, we am using this conversion in OT (e.g. [border routing](https://github.com/openthread/openthread/blob/ad91ac3b9c697357d1f89b806589e0533e2ed87a/src/core/border_router/routing_manager.hpp#L119-L122), srp [client](https://github.com/openthread/openthread/blob/ad91ac3b9c697357d1f89b806589e0533e2ed87a/src/core/net/srp_client.hpp#L492-L495)/[server](https://github.com/openthread/openthread/blob/ad91ac3b9c697357d1f89b806589e0533e2ed87a/src/core/net/srp_server.hpp#L488-L491) implementation).